src/lbaf/IO/lbsMeshWriter.py

```diff
@@ -0,0 +1,306 @@
+from logging import Logger
+import os
+import sys
+import math
+import numbers
+import random
+import vtk
+
+from .lbsGridStreamer import GridStreamer
+from ..Model.lbsPhase import Phase
+
+
+class MeshWriter:
+    """A class to write LBAF results to mesh files via VTK layer
+    """
+
+    def __init__(self, p: Phase, grid_size, object_jitter: float, f="lbs_out", r=1., output_dir=None, logger: Logger = None):
+        """ Class constructor:
+            p: Phase instance
+            grid_size: iterable containing grid sizes in each dimension
+            object_jitter: coefficient of random jitter with magnitude < 1
+            f: file name stem
+            r: grid_resolution value
+            output_dir: output directory
+        """
+        # Assign logger to instance variable
+        self.__logger = logger
+
+        # Ensure that provided phase has correct type
+        if not isinstance(p, Phase):
+            self.__logger.error("Could not write to ExodusII file by lack of a LBS phase")
+            sys.exit(1)
+        self.__phase = p
+
+        # Retrieve number of mesh points and bail out early if empty set
+        if not (n_p := len(self.__phase.get_ranks())):
+            self.__logger.error("Empty list of ranks, cannot write a mesh file")
+            return
+        self.__n_p = n_p
+
+        # Ensure that an iterable of grid sizes was passed
+        if not isinstance(grid_size, (list, tuple)):
+            self.__logger.error("Could not generate meshes by lack of a grid size iterable")
+            sys.exit(1)
+        self.__grid_size = grid_size
+        self.__logger.info(
+            f"Mapping {self.__n_p} ranks onto a {grid_size[0]}x"
+            f"{grid_size[1]}x{grid_size[2]} rectilinear grid")
+
+        # Ensure that a valid object jitter coefficient was passed
+        if not isinstance(object_jitter, float) or abs(object_jitter) >= 1.0:
+            self.__logger.error(f"Could not generate meshes due to unsuitable object jitter coefficient: {object_jitter}")
+            sys.exit(1)
+        self.__object_jitter = object_jitter
+        
+        # Ensure that specified grid resolution is correct
+        if not isinstance(r, numbers.Number) or r <= 0.:
+            self.__logger.error("Grid resolution must be a positive number")
+            sys.exit(1)
+        self.__grid_resolution = float(r)
+
+        # Assemble file and path names from constructor parameters
+        self.__rank_file_name = f"{f}_rank_view.e"
+        self.__object_file_name = f"{f}_object_view"
+        self.__output_dir = output_dir
+        if self.__output_dir is not None:
+            self.__rank_file_name = os.path.join(
+                self.__output_dir,
+                self.__rank_file_name)
+            self.__object_file_name = os.path.join(
+                self.__output_dir,
+                self.__object_file_name)
+
+
```

Comment:
Just one blank line. Two blank lines are expected between functions and classes and one blank line is expected between methods of a class.

Reply:
@ppebay